### PR TITLE
SDK-93 added support for openmrs-core development

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**

### CODE REVIEW

File: src/main/java/org/openmrs/maven/plugins/ModuleInstall.java [hunk 1]
```diff
@@ -114,20 +116,67 @@ public void installModule(String serverId, String groupId, String artifactId, St
...
+        if(artifact != null){
+            artifactItems.add(artifact.toElement(serverPath.toString()));
+
+            executeMojoPlugin(artifactItems);
+
+            File openmrsCorePath = new File(serverPath, "openmrs-" + properties.getOpenmrsCoreVersion() + ".war");
+            openmrsCorePath.delete();
+
+            properties.setParam("openmrs.platform.version", mavenProject.getVersion());
+            properties.save();
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: I'd put lines 119-131 in a separate method: boolean deployOpenmrsWar(...), which returns true if it was upgraded

File: src/main/java/org/openmrs/maven/plugins/ModuleInstall.java [hunk 2]
```diff
@@ -114,20 +116,67 @@ public void installModule(String serverId, String groupId, String artifactId, St
...
+            String moduleId = artifact.getArtifactId();
+            moduleId = moduleId.substring(0, moduleId.indexOf("-omod"));
+            String[] params = {artifact.getGroupId(), moduleId, artifact.getVersion()};
+            String module = StringUtils.join(params, "/");
+            properties.addToValueList(SDKConstants.PROPERTY_USER_MODULES, module);
+            properties.save();
+            if (moduleRemoved) {
+                getLog().info(String.format(DEFAULT_UPDATE_MESSAGE, moduleId, artifact.getVersion()));
+            }
+            else getLog().info(String.format(DEFAULT_OK_MESSAGE, moduleId));
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: I'd put lines 133-153 in a separate method deployModule()

File: src/main/java/org/openmrs/maven/plugins/ModuleInstall.java [hunk 3]
```diff
@@ -114,20 +116,67 @@ public void installModule(String serverId, String groupId, String artifactId, St
...
-        artifact.setArtifactId(artifact.getArtifactId() + "-omod");
-        File modules = new File(serverPath, SDKConstants.OPENMRS_SERVER_MODULES);
-        modules.mkdirs();
-        Element[] artifactItems = new Element[1];
-        artifactItems[0] = artifact.toElement(modules.getPath());
-        File[] listOfModules = modules.listFiles();
-        boolean versionUpdated = false;
-        boolean removed = false;
+        ServerConfig properties = ServerConfig.loadServerConfig(serverPath);
+        Artifact artifact = checkOpenmrsCoreVersion(helper, properties);
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: rename to checkCurrentDirectoryForOpenmrsCoreVersion

File: src/main/java/org/openmrs/maven/plugins/ModuleInstall.java (hunk 1 above)
Comment (AdamGrzybkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: @rkorytkowski 
I have updated PR

File: src/main/java/org/openmrs/maven/plugins/ModuleInstall.java [hunk 4]
```diff
@@ -114,20 +116,107 @@ public void installModule(String serverId, String groupId, String artifactId, St
...
+        artifactItems.add(artifact.toElement(serverPath.toString()));
+
+        executeMojoPlugin(artifactItems);
+
+        File openmrsCorePath = new File(serverPath, "openmrs-" + properties.getOpenmrsCoreVersion() + ".war");
+        openmrsCorePath.delete();
+
+        properties.setParam("openmrs.platform.version", mavenProject.getVersion());
+        properties.save();
+        return true;
```
Comment (rkorytkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: If you always return true then there is not point to return anything :) Please change to void. I know you went after my suggestion, but you implemented it a bit differently than I thought, which is also ok.

Comment (AdamGrzybkowski) [inline src/main/java/org/openmrs/maven/plugins/ModuleInstall.java]: @rkorytkowski  done

**Comment by AdamGrzybkowski** [comment]
@rkorytkowski I've resolved conflicts after last merge

**Pull request merged**